### PR TITLE
[Dialogs] Add dialog category for theming

### LINKED ISSUE #5836: [Dialogs] Internal issue: b/118160386
This was filed as an internal issue. If you are a Googler, please visit [b/118160386](http://b/118160386) for more details.

---
#### Internal data
- Associated internal bug: [b/118160386](http://b/118160386)
- Blocked by: https://github.com/material-components/material-components-ios/issues/5882

### PR DESCRIPTION
**Pull request by andrewoverton**
These changes add the Dialogs theming category.
Closes #5836.

### CODE REVIEW

File: MaterialComponentsAlpha.podspec [hunk 1]
```diff
@@ -79,6 +79,17 @@ Pod::Spec.new do |mdc|
...
   end
 
+  mdc.subspec "Dialogs+Theming" do |extension|
+    extension.ios.deployment_target = '8.0'
+    extension.public_header_files = "components/#{extension.base_name.split('+')[0]}/src/#{extension.base_name.split('+')[1]}/*.h"
+    extension.source_files = "components/#{extension.base_name.split('+')[0]}/src/#{extension.base_name.split('+')[1]}/*.{h,m}", "components/#{extension.base_name.split('+')[0]}/src/#{extension.base_name.split('+')[1]}/private/*.{h,m}"
+    extension.dependency "MaterialComponents/#{extension.base_name.split('+')[0]}"
+    extension.dependency "MaterialComponents/#{extension.base_name.split('+')[0]}+ColorThemer"
+    extension.dependency "MaterialComponents/#{extension.base_name.split('+')[0]}+TypographyThemer"
+    extension.dependency "MaterialComponents/ShadowElevations"
```
Comment (codeman7) [inline MaterialComponentsAlpha.podspec:89]: Do we need to depend on Button+Theming?

**Review by codeman7 — APPROVED** [review]

File: components/Dialogs/src/Theming/MDCAlertController+Theming.h [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+// limitations under the License.
+
+#import "MaterialContainerScheme.h"
+#import "MaterialDialogs.h"
+
+/**
+ This category is used to style MDCAlertController instances to a specific Material style which can
+ be found within the [Material Guidelines](https://material.io/design/components/buttons.html).
+ */
+@interface MDCAlertController (MaterialTheming)
```
Comment (galiak11) [inline components/Dialogs/src/Theming/MDCAlertController+Theming.h]: Tip: Should the file name match the category name? 
i.e.: MDCAlertController+MaterialTheming.h

File: components/Dialogs/tests/unit/DialogsThemingTests.swift [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+import XCTest
+
+import MaterialComponents.MaterialDialogs
+import MaterialComponents.MaterialColorScheme
+import MaterialComponents.MaterialTypographyScheme
+import MaterialComponents.MaterialShadowElevations
+import MaterialComponentsAlpha.MaterialDialogs_Theming
+
+class DialogsThemerTests: XCTestCase {
```
Comment (galiak11) [inline components/Dialogs/tests/unit/DialogsThemingTests.swift]: Can we also add a test for dialog-buttons? Verify that buttons are styled based on their emphases?

File: components/Dialogs/examples/DialogsAlertCustomizationViewController.swift [hunk 4]
```diff
@@ -57,14 +57,7 @@ class CustomAlertViewController: UIViewController {
...
 
-  var colorScheme = MDCSemanticColorScheme()
-  var typographyScheme = MDCTypographyScheme()
-  var alertScheme: MDCAlertScheme {
-    let scheme = MDCAlertScheme()
-    scheme.colorScheme = self.colorScheme
-    scheme.typographyScheme = self.typographyScheme
-    return scheme
-  }
+  var scheme: MDCContainerScheme = MDCContainerScheme()
```
Comment (galiak11) [inline components/Dialogs/examples/DialogsAlertCustomizationViewController.swift]: I believe all examples need to use the same name for the instance scheme - so it can be injected by our example app. Something like "containerScheme" or "materialScheme", etc would work better in that case.

File: components/Dialogs/examples/DialogsAlertViewController.m [hunk 5]
```diff
@@ -26,8 +26,6 @@
 
 @interface DialogsAlertViewController : MDCCollectionViewController
 @property(nonatomic, strong, nullable) NSArray *modes;
-@property(nonatomic, strong, nullable) MDCSemanticColorScheme *colorScheme;
```
Comment (galiak11) [inline components/Dialogs/examples/DialogsAlertViewController.m:29]: A materialScheme / containerScheme instance should replace the color/typography instances, so all examples can have consistent theming.
Make sure to set a default scheme in either the init method or in viewDidLoad (which I think is becoming the common place for that in our examples).

File: components/Dialogs/examples/DialogsTypicalUseViewController.m [hunk 6]
```diff
@@ -44,22 +41,6 @@ - (void)viewDidLoad {
     self.colorScheme =
```
Comment (galiak11) [inline components/Dialogs/examples/DialogsTypicalUseViewController.m:41]: Is there a reason to keep the color scheme?

File: MaterialComponentsAlpha.podspec (hunk 1 above)
Comment (andrewoverton) [reply MaterialComponentsAlpha.podspec:89]: Yes! Thanks

File: components/Dialogs/src/Theming/MDCAlertController+Theming.h (hunk 2 above)
Comment (andrewoverton) [reply components/Dialogs/src/Theming/MDCAlertController+Theming.h]: Yes, it should. We should fix this with buttons too...

File: components/Dialogs/examples/DialogsTypicalUseViewController.m (hunk 6 above)
Comment (andrewoverton) [reply components/Dialogs/examples/DialogsTypicalUseViewController.m:41]: It's used [elsewhere](https://github.com/material-components/material-components-ios/pull/5905/files#diff-31c3277d8e88d585f221a2c199ec8140R44) in the view controller so I decided to keep it

File: components/Dialogs/examples/DialogsAlertCustomizationViewController.swift (hunk 4 above)
Comment (andrewoverton) [reply components/Dialogs/examples/DialogsAlertCustomizationViewController.swift]: I'll rename it to `containerScheme` 👍

File: components/Dialogs/src/Theming/MDCAlertController+Theming.h [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MaterialContainerScheme.h"
+#import "MaterialDialogs.h"
+
+/**
+ This category is used to style MDCAlertController instances to a specific Material style which can
+ be found within the [Material Guidelines](https://material.io/design/components/buttons.html).
```
Comment (romoore) [inline components/Dialogs/src/Theming/MDCAlertController+Theming.h]: Drive-by nit: Looks like this is the wrong link to the guidelines article?

Suggestion: https://material.io/design/components/dialogs.html

Comment (andrewoverton) [reply components/Dialogs/src/Theming/MDCAlertController+Theming.h]: Thank you 🙏

File: components/Dialogs/tests/unit/DialogsThemingTests.swift (hunk 3 above)
Comment (andrewoverton) [reply components/Dialogs/tests/unit/DialogsThemingTests.swift]: Done.

File: components/Dialogs/examples/DialogsAlertViewController.m (hunk 5 above)
Comment (andrewoverton) [reply components/Dialogs/examples/DialogsAlertViewController.m:29]: done
